src/GoogleAdWordsEventForwarder.js

```diff
@@ -124,23 +120,46 @@
             return 'Can\'t send to forwarder ' + name + ', not initialized. Event added to queue.';
         }
 
-        function setEnhancedConversionData(enhancedConversionData) {
-            if (enhancedConversionData.email) {
-                window.enhanced_conversion_data.email = enhancedConversionData.email;
-            }
-            if (enhancedConversionData.phone_number) {
-                window.enhanced_conversion_data.phone_number = enhancedConversionData.phone_number;
-            }
-            if (enhancedConversionData.first_name) {
-                window.enhanced_conversion_data.first_name = enhancedConversionData.first_name;
-            }
-            if (enhancedConversionData.last_name) {
-                window.enhanced_conversion_data.last_name = enhancedConversionData.last_name;
-            }
-            if (enhancedConversionData.home_address) {
-                window.enhanced_conversion_data.home_address =
-                    enhancedConversionData.home_address;
+        function sanitizeEnhancedConversionData(conversionData) {
+            const sanitizedConversionData = {};
+
+            const allowedKeys = [
+                'email',
+                'phone_number',
+                'first_name',
+                'last_name',
+                'home_address',
+            ];
+
+            allowedKeys.forEach(function (key) {
+                if(conversionData[key]) {
+                    sanitizedConversionData[key] = conversionData[key];
+                }
+            });
+
+            return sanitizedConversionData;
+        }
```

Comment:
My opinion is that we can simplify this further and not even sanitize.  We just jump right to the `parseEnhancedConversionData` the `enhancedConversionData`.  My comment about how it seems Google updated `home_address` to be `address` and if we just pass through to Google what they pass to us, then the onus is on the developer to get the API right, and we don't have to update our code if Google adds any new keys.

Reply:
I can be convinced otherwise though.  But if you decide to sanitize, we have to warn them if there is a key that is not included so that they know there's a typo.  We either help them all the way (sanitize and inform of errors, which we do across our core SDK) or not at all (no sanitization, just pass to Google a parsed string).